### PR TITLE
Enable Hardware Crypto For nRF52840 NFC Example

### PR DESCRIPTION
**Pull request by felix-gohla**
HW crypto is automatically enabled with enabling the NRF security Kconfig. See README. :)

### CODE REVIEW

**Review by sirkrypt0 — APPROVED** [review]

**Comment by felix-gohla** [comment]
@sirkrypt0 Rebased and fixed some things.

File: examples/nfc_sim/nfc_simulator.c [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+        #define log printk
+    #elif defined(ESP_PLATFORM)
+        #define log printf
+    #else
+        #define log(...) do {} while(0);
+    #endif
+#endif
+
+static void *example_open() {
+    printf("open\n");
```
Comment (sirkrypt0) [inline examples/nfc_sim/nfc_simulator.c:27]: Those `printf` should all be `log`, no?

**Review by sirkrypt0 — APPROVED** [review]

**Comment by felix-gohla** [comment]
@sirkrypt0 Again rebased after merging #26.

File: examples/nrf52/src/main.c [hunk 2]
```diff
@@ -6,141 +6,29 @@
...
+        return -1;
+    } else {
+        printk("Initialized cryptography.\n");
     }
 
-    if (fido_dev_close(&dev) != FIDO_OK) {
-        return 4;
+    fido_dev_t dev;
+    if (prepare_stateless_rp_nfc_simulator_device(&dev) != 0) {
+        printk("Could not setup simulator device.\n");
```
Comment (sirkrypt0) [inline examples/nrf52/src/main.c:30]: Should we return here?

```suggestion
        printk("Could not setup simulator device.\n");
        return -1;
```

**Review by sirkrypt0 — CHANGES_REQUESTED** [review]

**Review by sirkrypt0 — APPROVED** [review]

**Pull request merged**